### DISCUSSION TITLE
Add docstring prompt

### DISCUSSION THREAD

**Discussion by tim-hilde** [olimorris/codecompanion.nvim#364] [Ideas]
It would be really nice to have the docstring generation feature of copilot integrated. I think the only tricky part ist referencing the current function and pasting the generated text at the correct position.

**Comment by olimorris** [comment]
Great idea. I would very much welcome this as a PR.

You could send the buffer as additional context to the LLM to make sure the doc string is correct.

**Reply by tim-hilde** [reply]
Great! Will see what I can do. Currently an issue is the positioning of the llm output. Is there a way to control the exact position?

**Reply by olimorris** [reply]
> Great! Will see what I can do. Currently an issue is the positioning of the llm output. Is there a way to control the exact position?

You can specify the placement in the opts of the prompt library. The positions are covered in the prompt library section of the README and possibly the RECIPES guide

**Reply by tim-hilde** [reply]
Quick question: how do I reference the user input via the inline user prompt? Also, can I easily change the name of the prompt? I would want the user to input the preferred style.

**Comment by tim-hilde** [comment]
Alright, this is added via https://github.com/olimorris/codecompanion.nvim/pull/390
